experiments/AMIP/modular/components/slab_utils.jl

```diff
@@ -1,19 +1,23 @@
-"""
-    SlabSimulation{P, Y, D, I}
-
-The simulation structure for slab models.
-"""
-struct SlabSimulation{P, Y, D, I}
-    params::P
-    Y_init::Y
-    domain::D
-    integrator::I
-end
-
 """
     get_slab_energy(slab_sim, T_sfc)
 
 Returns the internal energy per unit area of the slab.
 """
 get_slab_energy(slab_sim, T_sfc) =
     slab_sim.integrator.p.params.ρ .* slab_sim.integrator.p.params.c .* T_sfc .* slab_sim.integrator.p.params.h
+
+"""
+      dss!(Y::ClimaCore.Fields.FieldVector, p::NamedTuple, t)
+
+Computes the weighted direct stiffness summation and updates `Y` in place.
+In the case of a column domain, no dss operations are performed.
+"""
+function dss!(Y::ClimaCore.Fields.FieldVector, p::NamedTuple, t)
+    for key in propertynames(Y)
+        field = getproperty(Y, key)
+        buffer = get_dss_buffer(axes(field), p)
+        ClimaCore.Spaces.weighted_dss!(field, buffer)
```

Comment:
LGTM

Reply:
If we are calling `weighted_dss` internally, I would recommend prefixing `weighted` to this specialized function name!

Reply:
Is there a reason we are passing argument `t` if we are not using it inside the function?

Reply:
Sounds good, thanks (see comment above)

Reply:
If we are actually calling `weighted_dss` function inside, can we also clarify this in the calling function name! (e.g. `weighted_dss_slab1` or so. We currently support both pure `dss` and `weighted_dss` (which combines the effect of `M^-1 x dss`)in `ClimaCore.jl` and it would be good to disambiguate!

Reply:
Also, is there a reason we need to add a third argument/`_` here?

Reply:
Sounds good, changed to `weighted_dss_slab!` and merging. Thank you both!